src/cntrl.c

```diff
@@ -470,8 +470,32 @@ void cntrl_device_fini(struct cntrl_device *device)
 	}
 }
 
-void print_cntrl(struct cntrl_device *ctrl_dev)
+static void print_cntrl(struct cntrl_device *ctrl_dev)
 {
 		printf("%s (%s)\n", ctrl_dev->sysfs_path,
 			ctrl_type_str[ctrl_dev->cntrl_type]);
 }
+
+
+static void print_cntrl_json(struct cntrl_device *ctrl_dev)
+{
+	printf("{ \"controller\" : \"%s\" , \"type\" : \"%s\"}\n", ctrl_dev->sysfs_path,
+			ctrl_type_str[ctrl_dev->cntrl_type]);
+}
+
+typedef void (*print_controller)(struct cntrl_device *ctrl_dev);
+
+void print_controllers(enum output_format output)
```

Comment:
I think that we should use same approach as for slots, create a structure to fill like slot_response and then left printing details in ledctl.c. There will be no need to move enum output_format and struct record_func to utils. These are ledctl specyfic.